secure/storage/src/crypto_key_storage.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{error::Error, policy::Policy, storage::Storage, Value};
+use libra_crypto::ed25519::{compat, Ed25519PublicKey};
+use libra_crypto::PrivateKey;
+use rand::{rngs::OsRng, Rng, SeedableRng};
+
+/// CryptoKeyStorage represents a secret storage engine that offers support for generating, using
+/// and managing cryptographic keys securely. The API offered by CryptoKeyStorage is inspired by the
+/// 'Transit Secret Engine' provided by Vault: a production-ready storage engine for cloud
+/// infrastructures (e.g., see https://www.vaultproject.io/docs/secrets/transit/index.html).
+pub trait CryptoKeyStorage: Storage {
```

Comment:
I see what your concern was... my recommendation for this is you can still retain set and create but you introduce a new top level function in the storage trait called create/set_key and let's not be overly generic so it would only generate ed25519 keys. These create/set do not take in a value as it is implied that some smart utility will produce the key on the users behalf. Then you can have a default key generator for use in systems that do not have backends that generate keys. You cannot use polymorphism in Rust, so that's likely the most convenient fashion.

Your overall approach matches my original expectations, though I recognize there are a lot of nuances around how rust works.

Reply:
Thanks, David! As discussed offline, I've taken the simpler approach and merged everything into the Storage trait!